packages/webpack-cli/src/plugins/dotenv-webpack-plugin.ts

```diff
@@ -0,0 +1,200 @@
+import dotenv from "dotenv-defaults";
+import { Compiler, DefinePlugin } from "webpack";
+
+interface EnvVariables {
+  [key: string]: string;
+}
+
+interface Logger {
+  info(message: string): void;
+  warn(message: string): void;
+  error(message: string): void;
+}
+
+interface DotenvConfig {
+  paths?: string[];
+  prefixes?: string[];
+  systemvars?: boolean;
+  allowEmptyValues?: boolean;
+  expand?: boolean;
+  ignoreStub?: boolean;
+  defaults?: boolean | string;
+}
+
+type CodeValue = string;
+
+const interpolate = (env: string, vars: EnvVariables): string => {
+  const matches = env.match(/\$([a-zA-Z0-9_]+)|\${([a-zA-Z0-9_]+)}/g) || [];
+  matches.forEach((match) => {
+    env = env.replace(match, interpolate(vars[match.replace(/\$|{|}/g, "")] || "", vars));
+  });
+  return env;
+};
+
+const isMainThreadElectron = (target: string | undefined): boolean =>
+  !!target && target.startsWith("electron") && target.endsWith("main");
+
+export class Dotenv {
+  #logger!: Logger;
+  #config: DotenvConfig;
+  #inputFileSystem: any;
+  #cached: boolean;
+
+  constructor(config: DotenvConfig = {}) {
+    this.#config = {
+      paths: ["./.env"],
+      prefixes: ["process.env.", "import.meta.env."],
+      ...config,
+    };
+    this.#cached = false;
+  }
+
+  public apply(compiler: Compiler): void {
+    this.#inputFileSystem = compiler.inputFileSystem;
+    this.#logger = compiler.getInfrastructureLogger("dotenv-webpack-plugin");
+    compiler.hooks.thisCompilation.tap("dotenv-webpack-plugin", async (compilation) => {
+      const cache = compilation.getCache("dotenv-webpack-plugin-compiler");
+      const nodeEnv = process.env.NODE_ENV || compilation.options.mode;
+      if (this.#config.paths) {
+        this.#config.paths = [
+          ...this.#config.paths,
+          ...(nodeEnv === "test" ? [] : [".env.local"]),
+          `.env.${nodeEnv}`,
+          `.env.${nodeEnv}.local`,
+        ];
```

Comment:
It is invalid, it should be:
```
this.#config.paths = [
         ".env",
          ...(nodeEnv === "test" ? [] : [".env.local"]),
          `.env.${nodeEnv}`,
          `.env.${nodeEnv}.local`,
        ];
```

and with another order and put this in `constructor`